lib/minitest/minitest_reporter_plugin.rb

```diff
@@ -36,9 +36,9 @@ def total_count(options)
         filter = options[:filter] || '/./'
         filter = Regexp.new $1 if filter =~ /\/(.*)\//
 
-        Minitest::Runnable.runnables.map(&:runnable_methods).flatten.find_all { |m|
-          filter === m || filter === "#{self}##{m}"
-        }.size
+        Minitest::Runnable.runnables.map { |runnable|
+          runnable.runnable_methods.map { |m| "#{runnable}##{m}" }
```

Comment:
Am I right that the important change is replacing "#{self}##{m}" with "#{runnable}##{m}"?
If so why self was the problem?
Another question: the code also tried to match "m" now it doesn't. Why?

Reply:
In the original code (`Runnable.run`), the method was called on the `Runnable` child class, so `self` was the name of the derived class and yields the desired result when assembling the test name.

Here, the `total_count` method is called on the Reporter, so `self` is the name of the reporter. Assembling the complete name uses wrongly the reporter name so matching on the test's parent class did not work.

It would also mean that if ran with a filter like `/Reporter/`, the progress bar would show more tests than actually run, because all tests would match.

The fix is to map the methods to use each runnable name when joining the test name so the filter works on the parent class name as intended.

If I add a `p` to show the assembled method name, on the `spec_test.rb` file:

Before: `"#<Minitest::Reporters::DelegateReporter:0x007f863e392058>#test_0001_works"`
After: `"String::#length#test_0001_works"`

For the second question, I considered that if `filter` matches `m`, it will also match `#{self}##{m}`, so both checks seemed redundant. Furthermore, it would require to return both test and full name from the `map` loops and the extra and redundant data. Here, we're collecting the minimal amount of data without changing the behavior.

Reply:
Thank you for the explanation.  The change is more clear for me now.


> For the second question, I considered that if filter matches m, it will also match #{self}##{m}, so both checks seemed redundant. Furthermore, it would require to return both test and full name from the map loops and the extra and redundant data. Here, we're collecting the minimal amount of data without changing the behavior.

I think this will not be true if the patter try to match exact name (e.g. /^Pattern$/)

Reply:
That's correct, I forgot about this kind of filtering. Also, I was double wrong, as I don't need the bubble up the data so it doesn't add as much complexity as I initially thought.

I pushed a revision with the original filtering behavior. Let me know if you'd prefer have it squashed.